### PR TITLE
env: add rest-gateway

### LINKED ISSUE #88: Add support for REST gateway
## Is your feature request related to a problem? Please describe.
I'm always frustrated when nspcc-dev/neofs-testcases#723 can't be done because of missing testlib code.

## Describe the solution you'd like
Add support for the REST gw to the lib.

### PR DESCRIPTION
**Pull request by tatiana-nspcc**
The `REST_GW` class was added to `NeoFSEnv`, and uploading/downloading tests were also added.

Close #88.

### CODE REVIEW

File: src/neofs_testlib/env/env.py [hunk 1]
```diff
@@ -612,6 +619,7 @@ def start(self):
         self._generate_config()
         logger.info(f"Launching HTTP GW: {self}")
         self._launch_process()
+        logger.info(f"Launched HTTP GW: {self}")
```
Comment (tatiana-nspcc) [inline src/neofs_testlib/env/env.py:622]: @evgeniiz321 Could you please take a look at these changes? I'm not sure if you'll be happy with them, but to check the logs, I need to know the GW properties after _launch_process (for example, self.stderr is empty before). Also, any feedback is more than welcome, as this is my first attempt at doing something in Python.

**Review by evgeniiz321 — DISMISSED** [review]

File: src/neofs_testlib/env/env.py (hunk 1 above)
Comment (evgeniiz321) [reply src/neofs_testlib/env/env.py:622]: This is fine

File: src/neofs_testlib/env/env.py [hunk 2]
```diff
@@ -638,3 +646,74 @@ def _launch_process(self):
...
+    def _launch_process(self):
+        self.stdout = NeoFSEnv._generate_temp_file()
+        self.stderr = NeoFSEnv._generate_temp_file()
+        stdout_fp = open(self.stdout, "w")
+        stderr_fp = open(self.stderr, "w")
+        rest_gw_env = {}
+
+        for index, sn in enumerate(self.neofs_env.storage_nodes):
+            rest_gw_env[f"REST_GW_POOL_PEERS_{index}_ADDRESS"] = sn.endpoint
+            rest_gw_env[f"REST_GW_POOL_PEERS_{index}_WEIGHT"] = "0.2"
```
Comment (cthulhu-rider) [inline src/neofs_testlib/env/env.py:712]: can it be zero? if not, i'd leave some comment why exactly this value

File: src/neofs_testlib/env/env.py (hunk 1 above)
Comment (cthulhu-rider) [reply src/neofs_testlib/env/env.py:622]: doesn’t directly relate to adding REST, maybe a separate commit?

**Review by evgeniiz321 — APPROVED** [review]

File: src/neofs_testlib/env/env.py (hunk 2 above)
Comment (tatiana-nspcc) [reply src/neofs_testlib/env/env.py:712]: This value is used for both the S3 gateway and the HTTP gateway, so I am also using it. I am unsure about the origin of this idea.

Comment (roman-khimov) [reply src/neofs_testlib/env/env.py:712]: Leave it as is, these weights don't matter much for the setup we use and tests we have. Likely we'll deal with them when we'd like to test if they work.

**Review by cthulhu-rider — APPROVED** [review]

**Pull request merged**